src/main/java/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,47 @@
+package service.impl;
+
+import database.Storage;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import model.FruitTransaction;
+import service.ShopService;
+import strategy.OperationStrategy;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    public OperationStrategy getOperationStrategy() {
+        return operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> fruitTransactions) {
+        Map<String, Integer> toStorage = fruitTransactions.stream()
+                .collect(Collectors.toMap(
+                        FruitTransaction::getFruit,
+                        fruitTransaction -> operationStrategy.getOperationHandler(
+                                fruitTransaction.getOperation())
+                                .getQuantityToAdd(fruitTransaction),
+                        Integer::sum
+                ));
```

Comment:
Here you are creating a map of fruit transactions and then adding it to the storage. But the logic of this method is a bit confusing and it's not clear what the keys and values of this map represent. Try to make your code more self-explanatory, for example by using meaningful variable names. Also, it would be better to separate this logic into different methods to make it more readable.